### PR TITLE
#2003 - Make isposdef check optional in Ellipsoid constructor

### LINKED ISSUE #2003: Make isposdef check optional in Ellipsoid constructor
Use: `Ellipsoid(c, M, check_positive_definite=false)`. This came up recently when using for `M` an `MArray` (`isposdef` is not defined yet for such array type).

### PR DESCRIPTION
**Pull request by mforets**
Closes #2003.

### CODE REVIEW

File: src/Sets/Ellipsoid.jl [hunk 1]
```diff
@@ -56,39 +73,34 @@ julia> σ(ones(3), E)
...
 isoperationtype(::Type{<:Ellipsoid}) = false
 isconvextype(::Type{<:Ellipsoid}) = true
 
-# convenience constructor without type parameter
-Ellipsoid(c::AbstractVector{N}, Q::AbstractMatrix{N}) where {N<:AbstractFloat} =
-    Ellipsoid{N}(c, Q)
-
 # convenience constructor for ellipsoid centered in the origin
-Ellipsoid(Q::AbstractMatrix{N}) where {N<:AbstractFloat} =
-    Ellipsoid(zeros(N, size(Q, 1)), Q)
```
Comment (schillic) [inline src/Sets/Ellipsoid.jl]: This change is unrelated and has been done in #1946 already. It is not clear to me whether we want to have that. Can you comment in #1946?

Comment (mforets) [reply src/Sets/Ellipsoid.jl]: #1946 has other changes too; i didn't have time to consider them yet..

Comment (mforets) [reply src/Sets/Ellipsoid.jl]: But you're right; i shouldn't have removed that constructor. It's just weird to have `Ellipsoid{N}` in line 82  and then again `new{N}(...)`  below.

Comment (schillic) [reply src/Sets/Ellipsoid.jl]: > #1946 has other changes too

No, there are just some consequences. That is why I think we should discuss these changes before removing stuff. You can undo the changes here or we wait until we reached an agreement.
